### PR TITLE
Public fixes

### PR DESCRIPTION
**Pull request by saeedseyfi**
Changes:
- Fix a test: prevent `conn` to be defined globally
- Update `Makefile` help. Pass `sed` in Mac. E.g.`make release SED=gsed VERSION=1.2.14`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -43,25 +52,26 @@ doc:
 
 .PHONY: release
 release:
-	$(SED) -i 's/\"version\":\ \"[0-9]\+\.[0-9]\+\.[0-9]\+\"/\"version\":\ \"$(VERSION)\"/' package.json
-	$(SED)  -i "s/Unreleased/`date +%Y-%m-%d`/" CHANGELOG.md
+	$(SED) -i '' 's/\"version\":\ \"[0-9]\+\.[0-9]\+\.[0-9]\+\"/\"version\":\ \"$(VERSION)\"/' package.json
+	$(SED) -i '' 's/Unreleased/`date +%Y-%m-%d`/' CHANGELOG.md
```
Comment (jcbrand) [inline Makefile]: I don't understand the point of this change (and the similar ones below). It looks wrong to me and the documentation you linked to doesn't shed any light.

Perhaps you didn't set the VERSION when you called `make release`?

It should be called like this: `make release VERSION=1.3`

File: Makefile [hunk 2]
```diff
@@ -43,25 +52,26 @@ doc:
...
 	make dist
 	make doc
 
-.PHONE: dist
+.PHONEY: dist
```
Comment (jcbrand) [inline Makefile]: Should be PHONY, not PHONEY

File: Makefile (hunk 1 above)
Comment (saeedseyfi) [reply Makefile]: Excuse me @jcbrand, it seems I linked a wrong reference. [Here is the right one.](https://stackoverflow.com/a/2321958/2980087)
In MacOS the parameter `-i` is looking for an extension to create a backup before changes.

I hope my changes work in macOS and Linux both.

Comment (saeedseyfi) [reply Makefile]: I tested my changes on Ubuntu. Unfortunately it didn't work.
```
$ sed -i '' 's/hel/wel/' hello.c
sed: s/hel/wel/: No such file or directory
```

Please advise how should we support both platforms in this case?

Comment (jcbrand) [reply Makefile]: On Mac GNU sed can still be used.

`brew install gnu-sed`
`SED=gsed make release VERSION=1.3`

Perhaps you'd be wiling to mention this in the Makefile (in the `help` text)?

File: .gitignore [hunk 3]
```diff
@@ -12,3 +12,4 @@ doc
 *~
 .*.sw?
 doc-tmp
+.idea
```
Comment (jcbrand) [inline .gitignore]: I don't have a strong opinion about this being added here. However, you can also add it to a global .gitignore file in your home folder. Then you don't need it on a per-project basis.

File: Makefile (hunk 1 above)
Comment (saeedseyfi) [reply Makefile]: done

File: .gitignore (hunk 3 above)
Comment (saeedseyfi) [reply .gitignore]: You're right :) fixed

**Comment by saeedseyfi** [comment]
@jcbrand Please review my changes :)

File: Makefile [hunk 4]
```diff
@@ -20,9 +20,17 @@ all: doc $(STROPHE) $(STROPHE_MIN)
...
 	@echo ""
-	@echo " release       Prepare a new release of strophe.js. E.g. `make release VERSION=1.2.14`"
-	@echo " serve         Serve this directory via a webserver on port 8000."
-	@echo " stamp-npm     Install NPM dependencies and create the guard file stamp-npm which will prevent those dependencies from being installed again."
+	@echo " all         Update docs + build $(STROPHE) and $(STROPHE_MIN)"
+	@echo " doc         Update docs"
+	@echo " dist        Build $(STROPHE), $(STROPHE_MIN) and $(STROPHE_LIGHT)"
+	@echo " jshint      Check the code quality"
+	@echo " check       Build and run the tests"
+	@echo " jshint      Check code quality"
```
Comment (jcbrand) [inline Makefile]: "jshint" is mentioned twice.

**Comment by jcbrand** [comment]
Looking good, just one little thing :)

**Pull request closed without merging**